hbase-server/src/main/java/org/apache/hadoop/hbase/replication/BaseReplicationEndpoint.java

```diff
@@ -91,7 +92,7 @@ public WALEntryFilter getWALEntryfilter() {
         }
       }
     }
-    return filters.isEmpty() ? null : new ChainWALEntryFilter(filters);
+    return filters.isEmpty() ? null : new ChainWALEntryFilter(filters, this.replicationPeer);
```

Comment:
Is there some other way to do this that does not require adding ReplicationPeer to the method signature? 

Why not create a WALEntryFilter that drops empty cells? Then you can add it to the chain if you want and no further changes are needed.